### PR TITLE
feat: add repo recommendations to user profile

### LINKED ISSUE #802: Feature: Repo recommendations
### Type of feature

🍕 Feature

### Current behavior

When a user shows up on their profile, they see 0 contributions. It would be nice to show recommendations for projects to contribute to. This should appear like an ad for folks with no or low contributions. 

We can show some up-and-coming projects and/or sell this as space for companies to highlight projects that need contributions. 

### Suggested solution

The goal here is to give a user with no contributions something to contribute to. 

GitHub's home page example lists projects with details. I am open to ideas, but this is a good patter to call out recommendations for contributions unobtrusively. 

I think we may want to list repos similarly with some copy. 

"Want to contribute to more places? Check out these trending repositories."

Above can be a link to a video or blog post on how to contribute for the first time.  

Screen Shot 2023-01-26 at 6 39 10 AM

## Work

- [x] #1000
- [x] #1001

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR: 

- Adds recommendations tab to a users profile page
- implements repo recommendations with user's selected topics (interests)
- refactors tab styling to match the design
- updates recommendation components props and structure
- Adds ui flow for users without interest to add one/multiple topics to their interests

### Notes:
Users with one interest gets `opensauced/insights` repo as an additional recommendation

## Related Tickets & Documents
Closes #802

## Mobile & Desktop Screenshots/Recordings
image

image

image

image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by brandonroberts** [comment]
@OgDev-01 preview failed to build

**Comment by OgDev-01** [comment]
> @OgDev-01 preview failed to build

Fixed now 🍕

File: components/organisms/ContributorProfileTab/contributor-profile-tab.tsx [hunk 1]
```diff
@@ -74,23 +121,24 @@ const ContributorProfileTab = ({
...
-          className="data-[state=active]:border-sauced-orange data-[state=active]:border-b-2 text-2xl"
-          value="Contributions"
-          onClick={() => handleTabUrl("Contributions")}
-        >
-          Contributions
-        </TabsTrigger>
+        {tabLinks.map((tab) => (
+          <TabsTrigger
+            key={tab}
+            className="data-[state=active]:border-sauced-orange data-[state=active]:border-b-2 text-2xl data-[state=active]:font-bold data-[state=active]:text-transparent data-[state=active]:bg-clip-text data-[state=active]:bg-gradient-to-r data-[state=active]:from-[#EA4600] data-[state=active]:to-[#EB9B00]"
```
Comment (bdougie) [inline components/organisms/ContributorProfileTab/contributor-profile-tab.tsx:136]: This design was misinterpreted. Only Recommendations should have the gradient. This is meant to be a special tab and entice users without highlights and contributions to click and see.

Screen Shot 2023-05-15 at 9 18 52 AM

Comment (OgDev-01) [reply components/organisms/ContributorProfileTab/contributor-profile-tab.tsx:136]: Ok...

Comment (OgDev-01) [reply components/organisms/ContributorProfileTab/contributor-profile-tab.tsx]: reversed 👍

**Comment by bdougie** [comment]
Should also be gradient when not selected image

**Comment by brandonroberts** [comment]
Should `Recommendations` be displayed when I'm visiting another users' profile?

image

**Comment by bdougie** [comment]
> Should `Recommendations` be displayed when I'm visiting another users' profile?
> 
> 

No. That tab should only show on your profile and be unique to you.

File: components/organisms/ContributorProfileTab/contributor-profile-tab.tsx [hunk 2]
```diff
@@ -44,19 +55,56 @@ const ContributorProfileTab = ({
...
+      toast({ description: "An error occured!", variant: "danger" });
+    }
+  };
+
+  const getRepoFullNameByInterests = () => {
+    const repoFullNames = interests.map((interest) => {
+      return recommendations[interest as interestsType].toString();
+    });
+    setRecommendedRepos(repoFullNames);
+    console.log(recommendedrepos);
```
Comment (brandonroberts) [inline components/organisms/ContributorProfileTab/contributor-profile-tab.tsx:105]:
```suggestion
```

File: components/organisms/ContributorProfileTab/contributor-profile-tab.tsx [hunk 3]
```diff
@@ -17,9 +17,18 @@ import Button from "components/atoms/Button/button";
...
+import { useToast } from "lib/hooks/useToast";
 
 import PaginationResults from "components/molecules/PaginationResults/pagination-result";
 import Pagination from "components/molecules/Pagination/pagination";
+import DashContainer from "components/atoms/DashedContainer/DashContainer";
+import LanguagePill from "components/atoms/LanguagePill/LanguagePill";
+import RecommendedRepoCard from "components/molecules/RecommendedRepoCard/recommended-repo-card";
+import recommendations from "lib/utils/recommendations";
+import Title from "components/atoms/Typography/title";
+import clsx from "clsx";
```
Comment (brandonroberts) [inline components/organisms/ContributorProfileTab/contributor-profile-tab.tsx:31]: Move to top with third-party imports
```suggestion
import clsx from "clsx";
```

File: next-types.d.ts [hunk 4]
```diff
@@ -194,6 +196,24 @@ interface GhUser {
   readonly repos_url: string;
 }
 
+interface GhContributor {
```
Comment (brandonroberts) [inline next-types.d.ts:199]: Is this being used?

File: components/organisms/ContributorProfileTab/contributor-profile-tab.tsx [hunk 5]
```diff
@@ -44,19 +55,56 @@ const ContributorProfileTab = ({
...
+    if (data) {
+      toast({ description: "Updated successfully", variant: "success" });
+    } else {
+      toast({ description: "An error occured!", variant: "danger" });
+    }
+  };
+
+  const getRepoFullNameByInterests = () => {
+    const repoFullNames = interests.map((interest) => {
+      return recommendations[interest as interestsType].toString();
```
Comment (brandonroberts) [inline components/organisms/ContributorProfileTab/contributor-profile-tab.tsx:102]: Why use `toString` here? This should be able to handle multiple repos per interest for the future

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: next-types.d.ts (hunk 4 above)
Comment (OgDev-01) [reply next-types.d.ts:199]: no... Taking off right away

File: components/molecules/RecommendedRepoCard/recommended-repo-card.tsx [hunk 6]
```diff
@@ -3,88 +3,92 @@ import { BiGitPullRequest } from "react-icons/bi";
...
-  description: string;
-  issues: number;
-  stars: number;
-  pullRequests: number;
-  contributions: DbContribution[];
 }
 
-const RecommendedRepoCard = ({ fullname, description, issues, pullRequests, stars, contributions }: RecommendedRepoCardProps): JSX.Element => {
-
+const RecommendedRepoCard = ({ fullname, className }: RecommendedRepoCardProps): JSX.Element => {
```
Comment (brandonroberts) [inline components/molecules/RecommendedRepoCard/recommended-repo-card.tsx:18]: nit: consistency with most other usage

```suggestion
const RecommendedRepoCard = ({ fullName, className }: RecommendedRepoCardProps): JSX.Element => {
```

File: components/organisms/ContributorProfileTab/contributor-profile-tab.tsx [hunk 7]
```diff
@@ -137,7 +136,8 @@ const ContributorProfileTab = ({
             className={clsx(
               "data-[state=active]:border-sauced-orange data-[state=active]:border-b-2 text-2xl",
               tab === "Recommendations" &&
-                " data-[state=active]:font-bold data-[state=active]:text-transparent data-[state=active]:bg-clip-text data-[state=active]:bg-gradient-to-r data-[state=active]:from-[#EA4600] data-[state=active]:to-[#EB9B00]"
+                " data-[state=active]:font-bold data-[state=active]:text-transparent data-[state=active]:bg-clip-text data-[state=active]:bg-gradient-to-r data-[state=active]:from-[#EA4600] data-[state=active]:to-[#EB9B00]",
+              user && user.user_metadata.user_name !== login && tab === "Recommendations" && "hidden"
```
Comment (takanome-dev) [inline components/organisms/ContributorProfileTab/contributor-profile-tab.tsx:140]: The recommendation tab won't show up if I'm logged in but will if I'm not. You should also apply the `hidden` class if there is no user 🍕 

recom-2

**Review by bdougie — APPROVED** [review]

**Review by takanome-dev — APPROVED** [review]
Looks good :+1:

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**